packages/perseus/src/widgets/radio/__tests__/hash-shuffle.test.ts

```diff
@@ -0,0 +1,142 @@
+        const choices = createChoices();
+
+        const result1 = hashBasedShuffle(choices, `12345`);
+        const result2 = hashBasedShuffle(choices, `67890`);
+
+        expect(result1.map((c) => c.originalIndex)).not.toEqual(
+            result2.map((c) => c.originalIndex),
+        );
+    });
+
+    it("uses choice ID for shuffling when available", () => {
+        const choicesWithIds = [
+            createChoice("choice-0-unique1a", 0),
+            createChoice("choice-1-unique2b", 1),
+            createChoice("choice-2-unique3c", 2),
+        ];
+
+        const choicesWithDifferentIds = [
+            createChoice("choice-0-different1", 0),
+            createChoice("choice-1-different2", 1),
+            createChoice("choice-2-different3", 2),
+        ];
+
+        const result1 = hashBasedShuffle(choicesWithIds, `12345`);
+        const result2 = hashBasedShuffle(choicesWithDifferentIds, `12345`);
+        // Different IDs should produce different shuffle orders
+        expect(result1.map((c) => c.originalIndex)).not.toEqual(
+            result2.map((c) => c.originalIndex),
+        );
+    });
+
+    it("handles choices without IDs by falling back to originalIndex", () => {
+        const choicesWithoutIds: RadioChoiceWithMetadata[] = [
+            {originalIndex: 0, content: "First", correct: false},
+            {originalIndex: 1, content: "Second", correct: false},
+            {originalIndex: 2, content: "Third", correct: false},
+        ];
+
+        const result = hashBasedShuffle(choicesWithoutIds, `12345`);
+
+        // Should not throw and should return all choices
+        expect(result).toHaveLength(3);
+        expect(
+            result.every((choice) => choicesWithoutIds.includes(choice)),
+        ).toBe(true);
+    });
+
+    it("preserves all choices in the result", () => {
+        const choices = createChoices();
+        const result = hashBasedShuffle(choices, `12345`);
+
+        expect(result).toHaveLength(choices.length);
+
+        // Check that all original choices are present
+        choices.forEach((originalChoice) => {
+            expect(result).toContainEqual(originalChoice);
+        });
+    });
+
+    it("produces different orders than the original for most seeds", () => {
+        const choices = createChoices();
+        let differentOrderCount = 0;
+        const testSeeds = [`1`, `2`, `3`, `4`, `5`, `6`, `7`, `8`, `9`, `10`];
+
+        testSeeds.forEach((seed) => {
+            const result = hashBasedShuffle(choices, seed);
+            const originalOrder = [0, 1, 2, 3];
+            const shuffledOrder = result.map((c) => c.originalIndex);
+
+            if (!arraysEqual(originalOrder, shuffledOrder)) {
+                differentOrderCount++;
+            }
+        });
+
+        // Most seeds should produce a different order than the original
+        expect(differentOrderCount).toBeGreaterThan(testSeeds.length / 2);
```

Comment:
50% chance of shuffling seems low. Is there something we could do that ensures _some_ shuffling happened?